runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1051,7 +1534,15 @@ TR::Register *J9::Power::TreeEvaluator::awrtbarEvaluator(TR::Node *node, TR::Cod
    }
 
 TR::Register *J9::Power::TreeEvaluator::awrtbariEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-   {
+   {   
+   if(cg->comp()->getOption(TR_EnableFieldWatch) && !node->getSymbolReference()->getSymbol()->isArrayShadowSymbol()){
+      TR::Node *valueNode = NULL;
+      TR::TreeEvaluator::getIndirectWrtbarValueNode(cg, node, valueNode, false);
+      TR::Register *valueReg = cg->evaluate(valueNode);
+      TR::Register *sideEffectRegister = cg->evaluate(node->getThirdChild());
+      TR::TreeEvaluator::rdWrtbarHelperForFieldWatch(node, cg, sideEffectRegister, valueReg);
+   }
```

Comment:
We can improve the formatting here. The `{` after the `if` needs to be on a new line and spaced accordingly.